### ISSUE TITLE
Invalid handling of multiple headers of same name

### ISSUE THREAD

**Issue by mgawinec** [mrmike/Ok2Curl#25]
My request is having two headers of the same name:
```
Cookie: JSESSIONID=....
Cookie: SOMETOKEN=....; OTHERTOKEN=....
```
This is correct with respect to HTTP protocol. However, `CurlBuilder` will take only the last header, because it is using a map with unique keys for header names:

```java
    public CurlBuilder(Request request, long limit) {
       ...

        final Headers headers = request.headers();
        for (int i = 0; i < headers.size(); i++) {
            \\ MG: If there was already a header with same name it will be overwritten here
            this.headers.put(headers.name(i), headers.value(i));
        }
    }
```

**Comment by mgawinec** [comment]
I don't actually have time to setup all Android environment and prepare and verify pull request, so I'm pasting a fix here:

```java
package com.moczul.ok2curl;

import java.io.IOException;
import java.nio.charset.Charset;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import okhttp3.Headers;
import okhttp3.MediaType;
import okhttp3.Request;
import okhttp3.RequestBody;
import okio.Buffer;
import okio.BufferedSink;
import okio.Okio;

import static com.moczul.ok2curl.StringUtil.join;

public class CurlBuilder {

    private static final String FORMAT_HEADER = "-H \"%1$s:%2$s\"";
    private static final String FORMAT_METHOD = "-X %1$s";
    private static final String FORMAT_BODY = "-d '%1$s'";
    private static final String CONTENT_TYPE = "Content-Type";

    private final String url;
    private String method;
    private String contentType;
    private String body;
    private Map<String, List<String>> headers = new HashMap<>();

    public CurlBuilder(Request request) {
        this(request, -1L);
    }

    public CurlBuilder(Request request, long limit) {
        this.url = request.url().toString();
        this.method = request.method();
        final RequestBody body = request.body();
        if (body != null) {
            this.contentType = getContentType(body);
            this.body = getBodyAsString(body, limit);
        }

        final Headers headers = request.headers();
        for (int i = 0; i < headers.size(); i++) {
            List<String> headerValues = this.headers.get(headers.name(i));
            if (headerValues == null) {
                headerValues = new ArrayList<>();
                this.headers.put(headers.name(i), headerValues);
            }
            headerValues.add(headers.value(i));
        }
    }

    private String getContentType(RequestBody body) {
        final MediaType mediaType = body.contentType();
        if (mediaType != null) {
            return mediaType.toString();
        }

        return null;
    }

    private String getBodyAsString(RequestBody body, long limit) {
        try {
            final Buffer sink = new Buffer();

            final MediaType mediaType = body.contentType();
            final Charset charset = getCharset(mediaType);

            if (limit > 0) {
                final BufferedSink buffer = Okio.buffer(new LimitedSink(sink, limit));
                body.writeTo(buffer);
                buffer.flush();
            } else {
                body.writeTo(sink);
            }

            return sink.readString(charset);
        } catch (IOException e) {
            return "Error while reading body: " + e.toString();
        }
    }

    private Charset getCharset(MediaType mediaType) {
        if (mediaType != null) {
            return mediaType.charset(Charset.defaultCharset());
        }

        return Charset.defaultCharset();
    }

    public String build() {
        List<String> parts = new ArrayList<>();
        parts.add("curl");
        parts.add(String.format(FORMAT_METHOD, method.toUpperCase()));

        for (String key : headers.keySet()) {
            List<String> values = headers.get(key);
            for(String value : values) {
                final String headerPart = String.format(FORMAT_HEADER, key, value);
                parts.add(headerPart);
            }
        }

        if (contentType != null && !headers.containsKey(CONTENT_TYPE)) {
            parts.add(String.format(FORMAT_HEADER, CONTENT_TYPE, contentType));
        }

        if (body != null) {
            parts.add(String.format(FORMAT_BODY, body));
        }

        parts.add(url);

        return join(" ", parts);
    }
}
```

**Comment by mrmike** [comment]
Hi @mgawinec

Thanks for opening the issue. I will fix it.

**Comment by mgawinec** [comment]
@mrmike Please, let me know when you release the fix.

**Comment by mrmike** [comment]
Hi @mgawinec,

I've added fix for the issue in PR #27 
Please check the latest version of the library 
```
compile 'com.github.mrmike:ok2curl:0.3.0'
```

Let me know if you find any issues.
